src/engine/bufferscalers/rubberbandworkerpool.cpp

```diff
@@ -0,0 +1,33 @@
+#include "engine/bufferscalers/rubberbandworkerpool.h"
+
+#include <rubberband/RubberBandStretcher.h>
+
+#include "engine/engine.h"
+#include "util/assert.h"
+
+RubberBandWorkerPool::RubberBandWorkerPool(UserSettingsPointer pConfig)
+        : QThreadPool() {
+    bool multiThreadedOnStereo = pConfig &&
+            pConfig->getValue(ConfigKey(QStringLiteral("[App]"),
+                                      QStringLiteral("keylock_multithreading")),
+                    false);
+    m_channelPerWorker = multiThreadedOnStereo
+            ? mixxx::audio::ChannelCount::mono()
+            : mixxx::audio::ChannelCount::stereo();
+    DEBUG_ASSERT(0 == mixxx::kEngineChannelCount % m_channelPerWorker);
```

Comment:
NIT: [yoda](https://en.wikipedia.org/wiki/Yoda_conditions)
```suggestion
    DEBUG_ASSERT(mixxx::kEngineChannelCount % m_channelPerWorker == 0);
```